### PR TITLE
CVM Pipeline & TCB Value report workflow

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- Add an Azure Function to spawn CVMs. This is triggered by a webhook from GH.
- Add workflow to run azure-snp-report-verify tool and validate results on second VM. Runs 1x per week.

This allows us to get up-to-date information on the currently deployed security version numbers and idkeydigest on Azure CVMs.  
[Successful run](https://github.com/edgelesssys/constellation/actions/runs/3088923565).
### Related issue
- [AB#2413](https://dev.azure.com/Edgeless/Edgeless/_boards/board/t/Edgeless%20Team/Stories/?workitem=2413)

### CODE REVIEW

**Review by thomasten — COMMENTED** [review]
The workflow and the changes to the tool LGTM. I'm not capable to review the azure function.

File: .github/runners/azure-cvm/cvm-template.json [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+                        "provisionVMAgent": true,
+                        "patchSettings": {
+                            "patchMode": "ImageDefault",
+                            "assessmentMode": "ImageDefault"
+                        },
+                        "ssh": {
+                            "publicKeys": [
+                                {
+                                    "path": "/home/azureuser/.ssh/authorized_keys",
+                                    "keyData": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDbK3OgaBSc0hJWC3ZkC5Xnz4k+HP47QHZgGVW4zUEcmXfc/LZMiOOkBhk8h0xBoZ1FCUvuMvfdNoiuqKA05Ut199g+kzpkjdqCEoUP6hexSKPjC71ftKN1bJ0ERi5K558JHMDcZOKgMKhl8gFq2iAHgRKdC7JIXg6AKBhqwELNuhAfYnRW3HPItI1EoIwbBl5lxliLe2GQme+BmU7/UACI8DqDqeOzDeBcOpGWEoGs346n8OO82prOjwEkSg9Tuu2JLs1eI2iwhtoT/CmGqZQW0tlCAVC10lnBEDF1DHLctXAZ5XVkU9WVjMuUeHkbwmL136OlxGPaTSqJvanmtLEu2jAakojG3c3cIRvrE+ZeyTOxQ257CUhQqMCckgqHwtzxyOHkS3WLu8bCsV4x8H0Rpw77uETUFO0hf4txxtk1pEI6UyfeJmMzYMvgYRI7wWhQpp0YoRTZVGA3mu6sLY/6l9towaIe2tFrQFWHY0PEyNlm6m3Wa3V2miGRd/aZdwU= generated-by-azure"
```
[the remark this answers is not in the document]
Comment (derpsteb) [reply .github/runners/azure-cvm/cvm-template.json:205]: When removing the ssh key the vm can't be deployed as either ssh key access or username + pw needs to be set. I added the pubkey to the key vault that is in the same resource group as the template.

File: .github/runners/azure-cvm/cvm-template.json [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+            "location": "northeurope",
+            "sku": {
+                "name": "Standard",
+                "tier": "Regional"
+            },
+            "zones": [
+                "3"
+            ],
+            "properties": {
+                "ipAddress": "20.123.92.94",
```
[the remark this answers is not in the document]
Comment (derpsteb) [reply .github/runners/azure-cvm/cvm-template.json]: Removed the pub IP resource.

File: .github/runners/azure-cvm/cvm-template.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (daniel-weisse) [reply .github/runners/azure-cvm/cvm-template.json:230]: Can you set the ssh key as a github secret?
While this is just a public key, I think it is best if we keep such things out of our codebase.

File: .github/runners/azure-cvm/cvm-template.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (derpsteb) [reply .github/runners/azure-cvm/cvm-template.json]: Ok, removing the pub IP resource actually works. Seemed to be a transient issue with ubuntu package downloads.

**Review by malt3 — APPROVED** [review]
LGTM

File: .github/runners/azure-cvm/cvm-template.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (derpsteb) [reply .github/runners/azure-cvm/cvm-template.json]: Saved in key vault now. Deployed through parameter. Fetching from keyvault happens in cloud function.

**Pull request merged**